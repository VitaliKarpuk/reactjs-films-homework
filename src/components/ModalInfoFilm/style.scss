.movie-info {
  width: 100%;
  height: 100%;
  padding: 32px;
  background-color: rgba(0, 0, 0, 0.781);
  position: absolute;
  top: 0;
  opacity: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  border-radius: 15px;
  -webkit-border-radius: 15px;
  -moz-border-radius: 15px;
  -ms-border-radius: 15px;
  -o-border-radius: 15px;
  transition: all 0.3s;
  -webkit-transition: all 0.3s;
  -moz-transition: all 0.3s;
  -ms-transition: all 0.3s;
  -o-transition: all 0.3s;

  .movie-info__genre {
    font-size: 22px;
    letter-spacing: 1px;
    line-height: 33px;
    color: #47c3f0;
    font-weight: 500;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    padding: 0;
  }

  .movie-info__rating {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 30px 20px 0 20px;
    overflow: hidden;

    h3 {
      font-size: 28px;
      letter-spacing: 1px;
      line-height: 44px;
      color: #ffffff;
      font-weight: bold;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    span {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 60px;
      height: 36px;
      border-radius: 6px;
      border: 2px solid #49c1f0;
      font-size: 24px;
      line-height: 44px;
      color: #47c3f0;
      font-weight: 500;
      text-align: center;
    }
  }

  p {
    font-size: 26px;
    line-height: 43px;
    color: #ffffff;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .movie-info__close {
    position: absolute;
    left: 32px;
    top: 32px;
    width: 32px;
    height: 32px;

    &::before,
    &::after {
      position: absolute;
      left: 15px;
      content: " ";
      height: 33px;
      width: 2px;
      background-color: rgb(255, 255, 255);
    }

    &::before {
      transform: rotate(45deg);
      -webkit-transform: rotate(45deg);
      -moz-transform: rotate(45deg);
      -ms-transform: rotate(45deg);
      -o-transform: rotate(45deg);
    }

    &::after {
      transform: rotate(-45deg);
      -webkit-transform: rotate(-45deg);
      -moz-transform: rotate(-45deg);
      -ms-transform: rotate(-45deg);
      -o-transform: rotate(-45deg);
    }

    &:hover {
      cursor: pointer;
    }
  }
}
