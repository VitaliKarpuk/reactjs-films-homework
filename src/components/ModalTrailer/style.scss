.modal__trailer{
  position: fixed;
  min-width: 100vw;
  height: 100vh;
  background-color: #8a8a8a9e;
  top: 0;
  left: 0;
  z-index: 7;

  .modali__trailer_window{
    width: 80%;
    height: 70vh;
    margin: 0 auto;
    margin-top: 15vh;
    background-color: rgb(94, 94, 94);
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    div{
      height: 90%;
      display: flex;
      align-items: center;
      justify-content: center;
      background-color: aliceblue;
    }
    iframe{
      width: 100%;
      height: 100%;
      border: none;
    }
  }
}