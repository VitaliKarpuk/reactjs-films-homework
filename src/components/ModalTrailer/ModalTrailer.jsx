import React from "react";
import PropTypes from "prop-types";

import "./style.scss";

const ModalTrailer = ({ className, title, handleCloseTrailer }) => {
  return (
    <div className="modal__trailer">
      <div className="modali__trailer_window">
        <button className={className} onClick={handleCloseTrailer}>
          close
        </button>
        <div>
          <h2>{title}</h2>
        </div>
      </div>
    </div>
  );
};

ModalTrailer.propTypes = {
  handleBtnInfo: PropTypes.func,
  handleCloseTrailer: PropTypes.func,
  className: PropTypes.string,
  title: PropTypes.string
};

ModalTrailer.defaultProps = {
  handleBtnInfo: () => {},
  handleCloseTrailer: () => {},
  className: "",
  title: ""
};
export default ModalTrailer;
