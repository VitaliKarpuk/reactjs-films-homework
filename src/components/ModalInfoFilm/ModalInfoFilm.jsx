import React from "react";
import PropTypes from "prop-types";
import ModalTrailer from "../ModalTrailer/ModalTrailer";

import "./style.scss";

const ModalInfoFilm = ({
  title,
  rating,
  genreFilm,
  overview,
  handleCloseInfo,
  handleShowrailer
}) => {
  return (
    <div className="movie-info">
      <div className="movie-info__close" onClick={handleCloseInfo}></div>
      <div className="movie-info__rating">
        <h3>{title}</h3>
        <span>{rating}</span>
      </div>
      <p className="movie-info__genre">
        {genreFilm &&
          genreFilm.map((item, index) =>
            genreFilm[index + 1] ? `${item}, ` : `${item}`
          )}
      </p>
      <p>{overview}</p>
      <button className="btn__watch" onClick={handleShowrailer}>Watch Now</button>
    </div>
  );
};

ModalInfoFilm.propTypes = {
  title: PropTypes.string.isRequired,
  rating: PropTypes.number,
  genreFilm: PropTypes.array,
  overview: PropTypes.string,
  handleCloseInfo: PropTypes.func,
  handleShowrailer: PropTypes.func
};

ModalInfoFilm.defaultProps = {
  rating: 0,
  genreFilm: [],
  overview: "",
  handleCloseInfo: () => {},
  handleShowrailer: () =>{}
};

export default ModalInfoFilm;
