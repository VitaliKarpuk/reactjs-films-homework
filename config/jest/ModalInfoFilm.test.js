/* eslint-disable no-undef */
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
import ModalInfoFilm from '../../src/components/ModalInfoFilm/ModalInfoFilm';
import ModalTrailer from '../../src/components/ModalTrailer/ModalTrailer';

describe('Test ModalInfo component', () => {
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<ModalInfoFilm />);
    renderer.getRenderOutput();
    expect(result.type).toBe('div');
    expect(result.props.className).toBe('movie-info');
    expect(result.props.children.length).toBe(5);
    expect(result.props.children[0].type).toBe('div');
    expect(result.props.children[1].type).toBe('div');
    expect(result.props.children[2].type).toBe('p');
    expect(result.props.children[3].type).toBe('p');
    expect(result.props.children[4]).toMatchObject(<ModalTrailer />)
    expect(result).toMatchSnapshot();
  })
})