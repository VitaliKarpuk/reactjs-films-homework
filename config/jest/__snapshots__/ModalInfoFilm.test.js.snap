// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ModalInfo component renders correctly 1`] = `
<div
  className="movie-info"
>
  <div
    className="movie-info__close"
  />
  <div
    className="movie-info__rating"
  >
    <h3 />
    <span />
  </div>
  <p
    className="movie-info__genre"
  />
  <p />
  <ModalTrailer
    className="btn__watch"
    title="Watch Now"
  />
</div>
`;
